addons/mail/static/src/core/common/message_actions.js

```diff
@@ -97,6 +97,13 @@ messageActionsRegistry
         title: (component) => (component.state.showTranslation ? _t("Revert") : _t("Translate")),
         onClick: (component) => component.onClickToggleTranslation(),
         sequence: 100,
+    })
+    .add("copy", {
+        condition: (component) => component.copyable,
+        icon: "fa-link",
+        title: _t("Copy Message Link"),
+        onClick: (component) => component.onClickCopy(),
+        sequence: 110,
```

Comment:
Usually not good to refer on fields of component that were not defined by the action itself.
- `copyable` is pure data so this should not be concern of component. Ok to rely on component just to get the message, but that's it (`props.message.linkCopyable`)
- `onClickCopy` can be a method on model, so that this is not an implementation specific to `Message` component, e.g. `message.copyLink()`. Although message actions currently work only in `Message` component, it's not necessarily intended to be only used there. Thread actions work in both Discuss app and Chat Window, and it's not great if we have to define methods twice.